torch/utils/cpp_extension.py

```diff
@@ -1262,4 +1349,4 @@ def _join_cuda_home(*paths):
 
 
 def _is_cuda_file(path):
-    return os.path.splitext(path)[1] in ['.cu', '.cuh']
+    return os.path.splitext(path)[1] in ['.cu', '.cuh', '.hip']
```

Comment:
(Putting this comment here because github won't let me put it where I want it)

It looks like this PR adds support for compiling ahead-of-time cpp extensions with ROCM. Did you also want to add support for just-in-time cpp extensions? I didn't see any changes to that setup. The code for that is roughly here: https://github.com/pytorch/pytorch/blob/e24ce0e524ba88df6ddab487ad9039ea1f324e01/torch/utils/cpp_extension.py#L1156-L1169

Reply:
I think we want this eventually (and I briefly looked into it), but we could either include it in this PR or do it separately.

Reply:
In this PR, we enable ahead of time compilation for extensions using HIP with the legacy compilation way (not ninja). We will add just in time and ninja compilation separately

Reply:
Thanks for the clarification